perllib/Open311/PostServiceRequestUpdates.pm

```diff
@@ -182,6 +182,24 @@ sub process_update {
         return;
     }
 
+    # Comments are ordered randomly.
+    # Some cobrands/APIs do not handle ordering by age their end (e.g.
+    # Northumberland + Alloy) so we skip comment for now if an older unsent
+    # one exists for the problem. Otherwise an older update may overwrite a
+    # newer one in Alloy etc.
+    my @unsent_comments_for_problem
+        = $problem->comments->search(
+            { state => 'confirmed', send_state => 'unprocessed' }
```

Comment:
If you here, include a parameter of confirmed < the $comment->confirmed, then if you get any result you can skip, no need to fetch them all and then check.